File: WorkSphere.Server/Model/ProjectTask.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using WorkSphere.Model;

namespace WorkSphere.Server.Model
{
    public class ProjectTask : BaseEntity
    {
        [Required(ErrorMessage = "Task name is required")]
        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }
        public int? Order { get; set; }

        [Required]
        public int Duration { get; set; } // Duration in days

        [JsonConverter(typeof(JsonStringEnumConverter))]
        public Status Status { get; set; }

        // One-to-Many: A task is assigned to an employee
        public int? EmployeeID { get; set; }

        [JsonIgnore]
        public virtual Employee? Employee { get; set; }

        // One-to-Many: A task belongs to a project
        public int? ProjectID { get; set; }

        [JsonIgnore]
        public virtual Project? Project { get; set; }
    }
}

File: WorkSphere.Client/src/app/features/project-task/models/project-task.ts

export class ProjectTask {

  id: number;
  createdAt: Date;
  updatedAt: Date;
  name: string;
  description: string;
  order: number;
  status: number;
  employeeID: number;
  projectID: number;

  constructor(id: number, createdAt: Date, updatedAt: Date, name: string, description: string, order: number, status: number, employeeID: number, projectID: number) {
    this.id = id;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.name = name;
    this.description = description;
    this.order = order;
    this.status = status;
    this.employeeID = employeeID;
    this.projectID = projectID;
  }
}
